### DISCUSSION TITLE
Windows

### DISCUSSION THREAD

**Discussion by nkomarov** [rcaelers/workrave#586] [Q&A] [answered]
I use Work Rave on Windows 11 and see some bugs on my PC and would try writing a patch but I need some help setting it up from sources.

I have MS VS and Qt Dev Studio, Git... what else do I need to build it on my machine and try to debug?

Who builds it for Windows? Is there a short description on how to prepare the development environment?

**Comment by nkomarov** [comment]
Anybody there?

**Comment by rcaelers** [comment, marked as answer]
The easiest way is to install MSYS2. You need to install the following package (using pacman -S <packages>):
            mingw-w64-clang-i686-clang
            mingw-w64-clang-i686-cmake
            mingw-w64-clang-i686-lld
            mingw-w64-clang-i686-ninja
            mingw-w64-clang-x86_64-adwaita-icon-theme
            mingw-w64-clang-x86_64-boost
            mingw-w64-clang-x86_64-clang
            mingw-w64-clang-x86_64-cmake
            mingw-w64-clang-x86_64-gcc-compat
            mingw-w64-clang-x86_64-gtkmm3
            mingw-w64-clang-x86_64-lld
            mingw-w64-clang-x86_64-ninja
            mingw-w64-clang-x86_64-openssl
            mingw-w64-clang-x86_64-python
            mingw-w64-clang-x86_64-python-pip
            mingw-w64-clang-x86_64-spdlog
            mingw-w64-clang-x86_64-jq
            mingw-w64-clang-x86_64-uasm
            mingw-w64-clang-x86_64-rust

If you use vscode, you can rename **https://github.com/rcaelers/workrave/blob/main/.vscode/settings-msys2.json** to settings.json and compile from with vscode using cmake

Otherwise, you can start a clang64 MSYS terminal and run:
cmake.exe -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=<install-dir> -DWITH_TESTS=ON -DCODE_COVERAGE=OFF -DWITH_SBOM=OFF -DWITH_TRACING=ON -DWITH_UI=Gtk+3 -DWITH_CRASHPAD=OFF -DWITH_AUTO_UPDATE=OFF -DWITH_CORE_NEXT=OFF  -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_TOOLCHAIN_FILE=cmake/toolchains/msys2.cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=TRUE  -S<soure-dir> -<bin-dir> -G Ninja
cmake --build <bin-dir>

Let me know if you need further assistance.
